MVVMIOSTests/UserViewModelTests.swift

```diff
@@ -8,82 +8,57 @@
     }
 
-    func testTheUserListDoesNotPopulateIfTheResposneIsNot200() {
-        let networkingClient: MockNetworkClient = MockNetworkClient()
-        networkingClient.result = Result<Response, Error>.failure(NetworkError.invalidURL as Error)
-        sut = makeSut(usingClient: networkingClient)
-        let urlwhichDoesNotRespondWith200 = URL(string: "https://searchy.search.com")!
-        sut?.fetchFromNetwork(from: urlwhichDoesNotRespondWith200)
+    func testTheUserListIsNotPopulatedWhenTheResponseIsNotSuccessFul() {
+        // given
+        networkingClient.result = Result<[Response], Error>.failure(NetworkError.invalidURL as Error)
+        let validURl = URL(string: "https://search.search.com")!
+        // when
+        sut?.fetchFromNetwork(from: validURl)
+        // then
         XCTAssertTrue(sut?.usersList.count == 0)
     }
 
-//    func testTheUserListIsParsedProperly() {
-//        let sampleJson =    """
-//                            {
-//                                "name": "Test User"
-//                            }
-//                            """
-//        let jsonData = sampleJson.data(using: .utf8)!
-//        sut = makeSut(withResult: .success(jsonData))
-//        let expectation = expectation(description: "It gets appropriate data")
-//        let expectedServerResponse: [UserListViewModel] = [UserListViewModel(from: .init(name: "Test User"))]
-//        let url = URL(string: "https://a-valid-url.com")!
-//        sut?.fetchFromNetwork(from: url)
-//        expectation.fulfill()
-//        wait(for: [expectation], timeout: 0.5)
-//
-//        XCTAssertEqual(expectedServerResponse.count, sut?.usersList.count)
-//
-//    }
+    func testTheUserListsGetsPopulatedWhenThereIsACorrectResponse() {
+
+        networkingClient.result = Result<[Response], Error>.success([
+            .init(name: "A new response"),
+            .init(name: "A Second User")
+        ])
+
+        let validURl = URL(string: "https://search.search.com")!
 
-//    func testTheUserListGetsPopulated() {
-//        let sampleJson =    """
-//                            {
-//                                "name": "Test User"
-//                            }
-//                            """
-//        let jsonData = sampleJson.data(using: .utf8)!
-//        let networkingClient: MockNetworkClient = MockNetworkClient(withResult: .success(jsonData))
-//        sut =  UserViewModel(networkingService: networkingClient)
-//        let response: Response = .init(name: "Hell world")
-//        networkingClient.resultOne = Result<Response, Error>.success(response)
-//        let url = URL(string: "https://a-valid-url.com")!
-//        sut?.fetchData(from: url)
-//    }
+        sut?.fetchFromNetwork(from: validURl)
 
-    private func makeSut(usingClient: MockNetworkClient) -> UserViewModel {
-        return UserViewModel(networkingService: usingClient)
+        XCTAssertTrue(sut?.usersList.count == 2)
```

Comment:
Nice! Might be good to assert that they are the users we're expecting too, and in the right order.